vitrivr-engine-module-features/src/main/kotlin/org/vitrivr/engine/module/features/feature/ehd/EHDRetriever.kt

```diff
@@ -0,0 +1,42 @@
+package org.vitrivr.engine.module.features.feature.ehd
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.flow
+import org.vitrivr.engine.core.context.QueryContext
+import org.vitrivr.engine.core.features.AbstractRetriever
+import org.vitrivr.engine.core.math.correspondence.LinearCorrespondence
+import org.vitrivr.engine.core.model.content.element.ImageContent
+import org.vitrivr.engine.core.model.descriptor.vector.FloatVectorDescriptor
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.query.proximity.ProximityQuery
+import org.vitrivr.engine.core.model.retrievable.attributes.DistanceAttribute
+import org.vitrivr.engine.core.model.types.Value
+import org.vitrivr.engine.core.operators.retrieve.Retriever
+
+/**
+ * [Retriever] implementation for the [EHD] analyser.
+ *
+ * @see [EHD]
+ *
+ * @author Ralph Gasser
+ * @version 1.0.0
+ */
+class EHDRetriever(field: Schema.Field<ImageContent, FloatVectorDescriptor>, query: ProximityQuery<Value.FloatVector>, context: QueryContext) : AbstractRetriever<ImageContent, FloatVectorDescriptor>(field, query, context) {
+
+    companion object {
+        /** [LinearCorrespondence] for [EHDRetriever]. Maximum distance is taken from Cineast implementation. */
+        private val CORRESPONDENCE = LinearCorrespondence(16 / 4f)
```

Comment:
Soo, `4f`? 😜

Reply:
:-)

As written: It's been taken from the Cineast implementation :P

Reply:
yeah, but I don't think we need to necessarily backport sins of the past

Reply:
I agree. 

Frankly, I was assuming there to be some "educational" value in the notation. Sometimes I express thing like this myself to hint at how the value was derived.